lib/Backends/CPU/Transforms.cpp

```diff
@@ -93,14 +93,30 @@ bool CPUBackend::transformPostLowering(Function *F,
     }
     if (auto *MN = dyn_cast<MaxNode>(&node)) {
       if (auto *splat = dyn_cast<SplatNode>(MN->getLHS())) {
-        auto MSN = F->addNode(new CPUMaxSplatNode(MN->getName(), MN->getRHS(),
+        NodeValue MNInput = MN->getRHS();
+        // Rescale might have been sunk into MaxNode, pull it back.
+        if (MNInput.getType() != MN->getResult().getType()) {
```

Comment:
could you also add an 'assert' before doing a transformation that output of the splat has the same type as the other input to the max operator? (we need to ensure that scale and offset of LHS and RHS are the same for quantized numbers).

Also, you can/should "Rescale" only for quantized operators.